src/api/libcellml/component.h

```diff
@@ -74,16 +76,32 @@ class LIBCELLML_EXPORT Component: public ComponentEntity
     void addVariable(const VariablePtr &v);
 
     /**
-     * @brief Remove the variable with the given name from this component.
+     * @brief Remove the variable with the given @p name from this component.
      * Remove the variable with the given name from this component. If the
-     * name is not found then no variable is removed.
+     * name is not found then throw @c std::out_of_range.
      * @sa addVariable
-     * @overload
      * @param name The name of the variable to remove.
      */
     void removeVariable(const std::string &name);
 
     /**
+     * @brief Remove the variable by the given @p variable pointer from this component.
+     * Remove the variable with the given pointer from this component. If the
+     * variable is not found then no variable is removed.
```

Comment:
Is there a reason this is different from the "remove by name" case? Seems like it would be sensible to throw for both, as it's an indication of programmer error?